patchwork-networking/src/main/java/net/minecraftforge/fml/network/NetworkRegistry.java

```diff
@@ -0,0 +1,389 @@
+package net.minecraftforge.fml.network;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import net.minecraftforge.fml.network.event.EventNetworkChannel;
+import net.minecraftforge.fml.network.simple.SimpleChannel;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.PacketByteBuf;
+
+/**
+ * The network registry. Tracks channels on behalf of mods.
+ */
+public class NetworkRegistry {
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final Marker NETREGISTRY = MarkerManager.getMarker("NETREGISTRY");
+	/**
+	 * Special value for clientAcceptedVersions and serverAcceptedVersions predicates indicating the other side lacks
+	 * this channel.
+	 */
+	@SuppressWarnings("RedundantStringConstructorCall")
+	public static String ABSENT = new String("ABSENT \uD83E\uDD14");
```

Comment:
Why use `new String(text)` anyways? What's the point? Unless some exceedingly stupid ASM relies on some obscure feature, this should be fine. Also, it should probably be `final`. Thanks cpw...